pkg/logical-cluster-manager/cluster-manager-controller.go

```diff
@@ -0,0 +1,244 @@
+package clustermanager
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/logical-cluster"
+
+	clusterproviderclient "github.com/kcp-dev/edge-mc/cluster-provider-client"
+	cluster "github.com/kcp-dev/edge-mc/cluster-provider-client/cluster"
+	v1alpha1apis "github.com/kcp-dev/edge-mc/pkg/apis/edge/v1alpha1"
+	edgeclient "github.com/kcp-dev/edge-mc/pkg/client/clientset/versioned"
+	"github.com/kcp-dev/edge-mc/pkg/client/informers/externalversions/edge/v1alpha1"
+)
+
+const controllerName = "cluster-manager"
+
+type op string
+
+const (
+	opAdd    op = "Add"
+	opUpdate op = "Update"
+	opDelete op = "Delete"
+)
+
+type queueItem struct {
+	op  op
+	key any
+}
+
+type Controller struct {
+	kubeconfig       *string
+	ctx              context.Context
+	clusterclientset edgeclient.Interface
+	clusterInformer  v1alpha1.LogicalClusterInformer
+	queue            workqueue.RateLimitingInterface
+}
+
+func NewController(
+	kubeconfig *string,
+	ctx context.Context,
```

Comment:
If I understand correctly, in the Kube world, if not the go world, there is a strong preference to make the context be the first parameter.